### PR TITLE
feat: add texteditor with interface

### PR DESCRIPTION
**Pull request by baseplate-admin**
There could be minor styling issues. Take a look

### CODE REVIEW

**Review by tokitou-san — CHANGES_REQUESTED** [review]
Good work!
Here are some bugs :)

Line breaking is not working properly on preview tab

[Screencast from 2023-06-12 21-40-02.webm](https://github.com/tokitou-san/CoreProject-V3-UI/assets/114811070/c91a4526-e5c0-4f10-89ea-b3dbb2c2a4e4)

**Comment by baseplate-admin** [comment]
tracking : https://github.com/markedjs/marked/issues/190

File: src/lib/components/shared/markdown.svelte [hunk 1]
```diff
@@ -2,7 +2,7 @@
     import DOMPurify from "isomorphic-dompurify";
     import { marked } from "marked";
 
-    export let markdown = "";
+    export let markdown: string;
```
Comment (baseplate-admin) [inline src/lib/components/shared/markdown.svelte:5]: Dont change it. Marked throws error with empty string

File: src/lib/components/shared/text_editor.svelte [hunk 2]
```diff
@@ -7,7 +7,7 @@
     import Markdown from "./markdown.svelte";
 
     let textarea_element: HTMLTextAreaElement;
-    let textarea_value = "";
+    let textarea_value: string;
```
Comment (baseplate-admin) [inline src/lib/components/shared/text_editor.svelte:10]: Dont change it. Marked throws error with empty string

File: src/lib/components/shared/markdown.svelte [hunk 3]
```diff
@@ -11,6 +11,7 @@
         mangle: false, // Isn't supported by typescript
         pedantic: true,
         gfm: true,
+        breaks: true,
```
Comment (baseplate-admin) [inline src/lib/components/shared/markdown.svelte:14]: What is the point?

File: src/lib/components/shared/text_editor.svelte [hunk 4]
```diff
@@ -227,21 +231,49 @@
...
+                type="button"
+            >
+                <Italic class="h-[1.5vw] text-surface-200" />
+            </button>
+            <button
+                class="btn p-0"
+                type="button"
+            >
+                <Underline class="h-[1.35vw] text-surface-200" />
+            </button>
```
Comment (baseplate-admin) [inline src/lib/components/shared/text_editor.svelte:269]: Switch to better DS

**Pull request merged**